gremlin-go/driver/graphTraversal.go

```diff
@@ -0,0 +1,655 @@
+/*
+Licensed to the Apache Software Foundation (ASF) Under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package gremlingo
+
+type GraphTraversal struct {
+	t *Traversal
+}
+
+func NewGraphTraversal(graph *Graph, traversalStrategies *TraversalStrategies, bytecode bytecode) *GraphTraversal {
+	return &GraphTraversal{t: &Traversal{
+		graph:               graph,
+		traversalStrategies: traversalStrategies,
+		bytecode:            bytecode,
+	}}
+}
+
+// Clone make a copy of a traversal that is reset for iteration.
+func (g *GraphTraversal) Clone() *GraphTraversal {
```

Comment:
Not sure if I would consider this a clone. Clone to me would require deep copy, this is just a different pointer with the same underlying memory.